File: MS.NET/Fundamentals/Generics/Basics/genmethodtest.cs
using System;

static class Program
{

	private static T Select<T>(int selector, T first, T second)
	{
		if(selector < 0)
			return first;
		return second;
	}
	
	public static void Main(string[] args)
	{
		int s = args.Length > 0 ? int.Parse(args[0]) : 0;
			
		string ss = Select(s, "monday", "tuesday");
		Console.WriteLine("Selected string = {0}", ss);

		double sd = Select(s, 4.5, 3.4);
		Console.WriteLine("Selected double = {0}", sd);

		int si = Select(s, 123, 0xABC);
		Console.WriteLine("Selected int = {0}", si);


	}
}

File: Java/Fundamentals/Generic/Basics/GenMethodTest.java
class GenMethodTest{

	private static<T> T select(int sign, T first, T second){
		if(sign < 0)
			return first;
		return second;
	}

	public static void main(String[] args){
		int s = Integer.parseInt(args[0]);
		String ss = select(s, "monday", "tuesday"); 
		System.out.printf("Selected String = %s%n", ss);
		double sd = select(s, 3.40, 2.30); // automatic boxing(arguments) and unboxing(result)
		System.out.printf("Selected double = %s%n", sd);
		int si = select(s, 1234, 0xABCD);
		System.out.printf("Selected int = %s%n", si);
	}
}

